### DISCUSSION TITLE
Zephyr OS, boards with different performance and 6LoWPAN frame retransmission

### DISCUSSION THREAD

**Discussion by ValNyrov** [openthread/openthread#9060] [Q&A]
Hello OpenThread community!

My hardware configuration is:

1)One board nRF52840-DK acting as router and commissioner,OS Zephyr 3.3.99,chip speed 64 MHz
2)One board TI CC2652-launchpad, acting as minimal end divice and joiner,OS Zephyr 3.3.99,chip speed 48 MHz
3)Packet sniffer: Wireshark + nrf52840-dongle

In general, everything works and the commissioning process is completed.
But I noticed in the log of the nRF52840-DK board that in the commissioning process there are repetitions of sending some frames, it is highlighted in the screenshot with a red line. This is also visible in the wireshark log (lines 33-50).
![nRF52840_log](https://github.com/openthread/openthread/assets/111975619/20a93c8f-1880-4009-b90f-3210de3cac58)

Screenshot TI CC2652 log:
![tiCC2652_log](https://github.com/openthread/openthread/assets/111975619/3ac5a38a-719b-4817-aeaa-0cf866e324e3)

Full logs in attachment.

After the completion of the commissioning, I execute the ping command whith 1024 bytes on the side of the nRF52840-DK board, and there are also retries of transmission of some frames.

Since these boards have different chips with different speeds, I assume that the CC2652 board does not always have time to process a frame (this is just my guess).

Perhaps I should correctly set the timeouts for waiting ACK frame from CC2652-launchpad and increase the time between retransmissions of frames? Please tell me how to do it right.

Zephyr logs:
[nRF52840.log](https://github.com/openthread/openthread/files/11491188/nRF52840.log)
[ti_cc2652.log](https://github.com/openthread/openthread/files/11491189/ti_cc2652.log)

Wireshark:
[wireshark.pcapng.tar.gz](https://github.com/openthread/openthread/files/11491194/wireshark.pcapng.tar.gz)

What commands in Zephyr did I execute:
[zephyr_commands_nRF52840.txt](https://github.com/openthread/openthread/files/11491201/zephyr_commands_nRF52840.txt)
[zephyr_commands_ti_CC2652.txt](https://github.com/openthread/openthread/files/11491202/zephyr_commands_ti_CC2652.txt)

Best Regards,
Valeriy

**Comment by jwhui** [comment]
Looking at the pcap file, it does appear that many packets sent from nRF52840 to CC2652 do not have a corresponding ack message. I would suggest reaching out to TI for support on this.

**Reply by ValNyrov** [reply]
Thanks for the answer. What timeout value should be for corresponding ack message? Is the timeout value set in the driver or in the OpenThread code?

Best Regards,
Valeriy

**Reply by jwhui** [reply]
In the case of nRF52840, the ACK timeout is implemented in the radio driver.

However, the packet capture does not show ACKs being sent by CC2652 in many cases, so I don't think this is a timeout configuration issue.

**Reply by ValNyrov** [reply]
Thanks for the help Jonathan!

I decided to check how two nrf52840 boards will behave, one board is configured as FTD and the other as MTD.
Both boards are on my table at a distance of 50cm from each other, the signal level is about -45 dBm.
On the first board (FTD), I launched a ping (20 packets of 1024 bytes of data), and also saw in the logs that not all requests have a corresponding ack message. Is it normal that there are not always ack messages? Maybe something else needs to be configured?

wireshark:
[wireshark.pcapng.gz](https://github.com/openthread/openthread/files/11517178/wireshark.pcapng.gz)

Zephyr logs:
[nRF52840_FTD.txt](https://github.com/openthread/openthread/files/11517180/nRF52840_FTD.txt)
[nRF52840_MTD.txt](https://github.com/openthread/openthread/files/11517182/nRF52840_MTD.txt)


Best Regards,
Valeriy

**Comment by maciejbaczmanski** [comment]
Hi @ValNyrov. 
How do you build SW for nRF52840?
If you are using nRF Connect SDK, could you provide tag / commit sha used while building?
Also are you using any custom configuration for samples?

Best Regards,
Maciej
